lib/redis/lock.rb

```diff
@@ -38,18 +38,19 @@ def lock(&block)
       try_until_timeout do
         expiration = generate_expiration
         # Use the expiration as the value of the lock.
-        break if redis.setnx(key, expiration)
+        break if redis.setnx(key, expiration) && Redis.next.setnx(key, expiration)
```

Comment:
Something like this:
      try_until_timeout do
        expiration = generate_expiration
        # Use the expiration as the value of the lock.
        old_redis = redis.setnx(key, expiration)
        new_redis = Redis.next.setnx(key, expiration)
        break if old_redis && new_redis
        if old_redis ^ new_redis
          redis.del(key)
          Redis.next.del(key)
          next
        end

Which will retry to get both until timeout.